core/labeled-replies-3.3.md

```diff
@@ -0,0 +1,36 @@
+---
+title: IRCv3.3 Labeled replies
+layout: spec
+copyrights:
+  -
+    name: "Alexey Sokolov"
+    period: "2015"
+    email: "alexey-irc@asokolov.org"
+---
+Name of this capability is `labeled-replies`.
+If client requests `labeled-replies`, it MUST also request `batch` capability.
+
+If the capability is enabled, client MAY tag its messages with `label` tag.
+
+For messages from client which are tagged with `label`, server MUST send exactly
+one message tagged with the same `label` value in response to such message.
+If response contains more than one message, `batch` can be used to group them.
+In that case start of the batch MUST be tagged with the label, and the batch type MUST be `labeled-response`.
+
+If the message doesn't require any response, an empty batch MUST be sent.
+
+## Examples
+
+1. `echo-message`
+
+    Client: @label=pQraCjj82e PRIVMSG #channel :\x02Hello!\x02
+    Server: @label=pQraCjj82e :nick!user@host PRIVMSG #channel :Hello!
+
+2. `WHOIS`
+
+    Client: @label=mGhe5V7RTV WHOIS nick
+    Server: @label=mGhe5V7RTV :irc.example.com BATCH +NMzYSq45x labeled-response
+    Server: @batch=NMzYSq45x :irc.example.com 311 client nick ~ident host * :Name
+    ...
+    Server: @batch=NMzYSq45x :irc.example.com 318 client nick :End of /WHOIS list.
```

Comment:
On the other hand...
When response contains single line, it's not required to be in a batch at
all, so the batch type isn't necessary.
So just whois-without-numeric-end should probably be enough, without need
for nested batch.

2015-07-13 8:42 GMT+01:00 Mantas Mikulėnas <notifications@github.com>:

> In core/labeled-replies-3.3.md
> <https://github.com/ircv3/ircv3-specifications/pull/162#discussion_r34438927>
> :
>
> > +If the message doesn't require any response, an empty batch MUST be sent.
> > +
> > +## Examples
> > +
> > +1. `echo-message`
> > +
> > +    Client: @label=pQraCjj82e PRIVMSG #channel :\x02Hello!\x02
> > +    Server: @label=pQraCjj82e :nick!user@host PRIVMSG #channel :Hello!
> > +
> > +2. `WHOIS`
> > +
> > +    Client: @label=mGhe5V7RTV WHOIS nick
> > +    Server: @label=mGhe5V7RTV :irc.example.com BATCH +NMzYSq45x labeled-response
> > +    Server: @batch=NMzYSq45x :irc.example.com 311 client nick ~ident host * :Name
> > +    ...
> > +    Server: @batch=NMzYSq45x :irc.example.com 318 client nick :End of /WHOIS list.
>
> Well yes, but without BATCH, how would you know where a @label=asdfg
> series ends?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ircv3/ircv3-specifications/pull/162/files#r34438927>.
>